Source/ARTPush.m

```diff
@@ -225,7 +225,9 @@ + (void)didFailToRegisterForRemoteNotificationsWithError:(NSError *)error rest:(
 
 - (void)activate {
 #if TARGET_OS_IOS && !TARGET_OS_SIMULATOR
-    [[UIApplication sharedApplication] registerForRemoteNotifications];
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Consider check if the `currentThread` isn't the `mainThread` already.
Your code will work, but won't be called immediately.

Reply:
ok, will do

Reply:
https://github.com/ably/ably-cocoa/pull/1293/commits/a9e92dec5885775bb96509c82e2c65f8c8303a9e